browser/resources/settings/brave_wallet_page/brave_wallet_page.ts

```diff
@@ -136,6 +157,24 @@ class SettingsBraveWalletPage extends SettingsBraveWalletPageBase {
       { value: 'XDR' }
     ]
     this.currency_list_.every((x) => x.name = x.value);
+    window.addEventListener('load', this.onLoad_.bind(this));
```

Comment:
Why do we need to wait for page load or only run some code at page load? I worry that it's flakey, because there's not much that happens for settings at page load time. None of the chrome settings components do this.

Reply:
browserProxy_ is not ready as i remember until page is loaded

Reply:
Actually prefs are not accessible before

Reply:
Since you have an observer on prefs.brave.wallet.auto_pin_enabled.value, does it not get called when prefs is set if it isn't already? I'm sticking with thinking it is a good idea to not rely on `onLoad` because none of the Chromium settings page do that and they all rely on `this.prefs`.

Reply:
Fixed